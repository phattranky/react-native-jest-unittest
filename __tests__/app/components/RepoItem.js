import React from 'react';
import renderer from 'react-test-renderer';

import RepoItem from '../../../app/components/RepoItem';
import { repos } from '../../../mockData/repos';

it('renders a RepoItem using Snapshots', () => {
  expect(renderer.create(
    <RepoItem
      repo={repos[0]}
      selectRepo={jest.fn()}
    />
  )).toMatchSnapshot();
});

it('renders a selected RepoItem using Snapshots', () => {
  expect(renderer.create(
    <RepoItem
      isSelected
      repo={repos[0]}
      selectRepo={jest.fn()}
    />
  )).toMatchSnapshot();
});
