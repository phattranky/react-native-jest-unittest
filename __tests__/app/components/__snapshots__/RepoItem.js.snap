exports[`test renders a RepoItem using Snapshots 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}>
  <View
    style={
      Object {
        "paddingHorizontal": 24,
        "paddingVertical": 8,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "paddingBottom": 4,
        }
      }>
      Test 1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "paddingBottom": 8,
        }
      }>
      50000 stars
    </Text>
  </View>
</View>
`;

exports[`test renders a selected RepoItem using Snapshots 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}>
  <View
    style={
      Array [
        Object {
          "paddingHorizontal": 24,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#B2DFDB",
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "paddingBottom": 4,
        }
      }>
      Test 1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "paddingBottom": 8,
        }
      }>
      50000 stars
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Useful description goes here.
    </Text>
  </View>
</View>
`;
